symengine/pow.cpp

```diff
@@ -311,29 +323,52 @@ RCP<const Basic> Log::create(const RCP<const Basic> &a) const
 
 RCP<const Basic> log(const RCP<const Basic> &arg)
 {
-    if (eq(*arg, *zero)) {
-        throw NotImplementedError(
-            "log(0) is complex infinity. Yet to be implemented");
-    }
+    if (eq(*arg, *zero))
+        return ComplexInf;
     if (eq(*arg, *one))
         return zero;
     if (eq(*arg, *E))
         return one;
+    if (eq(*arg, *Inf))
+        return Inf;
+    if (eq(*arg, *NegInf))
+        return Inf;
+    if (eq(*arg, *ComplexInf))
+        return ComplexInf;
+
     if (is_a_Number(*arg)) {
         RCP<const Number> _arg = rcp_static_cast<const Number>(arg);
         if (not _arg->is_exact()) {
             return _arg->get_eval().log(*_arg);
         } else if (_arg->is_negative()) {
-            throw NotImplementedError(
-                "Imaginary Result. Yet to be implemented");
+            return add(log(mul(minus_one, _arg)), mul(pi, I));
         }
     }
+
     if (is_a<Rational>(*arg)) {
         RCP<const Integer> num, den;
         get_num_den(down_cast<const Rational &>(*arg), outArg(num),
                     outArg(den));
         return sub(log(num), log(den));
     }
+
+    if (is_a<Complex>(*arg)) {
```

Comment:
No..In ```is_a_Number``` It Checks whether argument is negative or not.Like if arg is -3(integer) then check results true and outputs log(3)+I*pi,But if arg is of form 3i or -3i , is_negative() function returns false and is left as log(3*I) / log(-3I) .In Such cases this if-block get executed and outputs log(3)+I*pi/2